nemo/core/classes/mixins/adapter_mixins.py

```diff
@@ -159,6 +182,11 @@ def add_adapter(self, name: str, cfg: DictConfig):
             cfg: A DictConfig or Dataclass that contains at the bare minimum `__target__` to instantiate a
                 new Adapter module.
         """
+        if self.get_accepted_adapter_types():
```

Comment:
Check returned value is not none, and isinstance (Iterable) and finally len(output) > 0 before entering. This current check is very risky for older models